File: Source/Padutronics.Windows.Win32.Api/DWrite/IDWriteTextFormat.cs
using Padutronics.Windows.Win32.Api.Unknwn;
using System.Runtime.InteropServices;
using System.Text;

namespace Padutronics.Windows.Win32.Api.DWrite;

[Guid("9c906818-31d7-4fd3-a151-7c5e225db55a")]
public interface IDWriteTextFormat : IUnknown
{
    void GetFontFamilyName(StringBuilder fontFamilyName, int nameSize);
    int GetFontFamilyNameLength();
    float GetFontSize();
    DWRITE_FONT_STRETCH GetFontStretch();
    DWRITE_FONT_STYLE GetFontStyle();
    DWRITE_FONT_WEIGHT GetFontWeight();
}

File: Source/Padutronics.Windows.Win32.Api.CWrapper/DWrite/IDWriteTextFormat.cpp
#include "IDWriteTextFormat.hpp"

auto IDWriteTextFormat_GetFontFamilyName(IDWriteTextFormat& _this, WCHAR* fontFamilyName, UINT32 nameSize) -> HRESULT
{
    return _this.GetFontFamilyName(fontFamilyName, nameSize);
}

auto IDWriteTextFormat_GetFontFamilyNameLength(IDWriteTextFormat& _this) -> UINT32
{
    return _this.GetFontFamilyNameLength();
}

auto IDWriteTextFormat_GetFontSize(IDWriteTextFormat& _this) -> FLOAT
{
    return _this.GetFontSize();
}

auto IDWriteTextFormat_GetFontStretch(IDWriteTextFormat& _this) -> DWRITE_FONT_STRETCH
{
    return _this.GetFontStretch();
}

auto IDWriteTextFormat_GetFontStyle(IDWriteTextFormat& _this) -> DWRITE_FONT_STYLE
{
    return _this.GetFontStyle();
}

auto IDWriteTextFormat_GetFontWeight(IDWriteTextFormat& _this) -> DWRITE_FONT_WEIGHT
{
    return _this.GetFontWeight();
}